src/jit/PerfDump.cpp

```diff
@@ -310,12 +310,12 @@ uint32_t PerfDump::dumpByteCode(InstructionListItem* item)
             fprintf(m_sourceFile, "%c R%d", prefix, static_cast<int>(i - paramCount));
         }
 
-        switch (VARIABLE_TYPE(operand->ref)) {
-        case Operand::Immediate:
+        switch (VARIABLE_TYPE(*operand)) {
+        case Instruction::Immediate:
```

Comment:
I think this is ConstPtr

https://github.com/Samsung/walrus/blob/main/src/jit/Compiler.h#L205

Did you tested the patch?

Reply:
Yes, this failed, I'll fix it